sharepoint-adfs/azuredeploy.json

```diff
@@ -0,0 +1,794 @@
+      }
+    },
+    "_artifactsLocation": {
+      "type": "string",
+      "metadata": {
+        "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
+      },
+      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/sharepoint-adfs/"
+    }
+  },
+  "variables": {
+    "generalSettings": {
+      "storageAccountName": "[concat('storage', uniqueString(resourceGroup().id))]",
+      "vmsStorageAccountContainerName": "vhds",
+      "dscScriptsFolder": "dsc",
+      "vmDCName": "DC",
+      "vmSQLName": "SQL",
+      "vmSPName": "SP"
+    },
+    "vNetSettings": {
+      "vNetPrivateName": "[concat(resourceGroup().name, '-vnet')]",
+      "vNetPrivatePrefix": "10.0.0.0/16",
+      "vNetPrivateSubnet1Name": "Subnet-1",
+      "vNetPrivateSubnet1Prefix": "10.0.1.0/24",
+      "vNetPrivateSubnet2Name": "Subnet-2",
+      "vNetPrivateSubnet2Prefix": "10.0.2.0/24",
+      "vNetPrivateSubnet3Name": "Subnet-3",
+      "vNetPrivateSubnet3Prefix": "10.0.3.0/24"
+    },
+    "vmDC": {
+      "vmImagePublisher": "MicrosoftWindowsServer",
+      "vmImageOffer": "WindowsServer",
+      "vmImageSKU": "2016-Datacenter",
+      "vmOSDiskName": "vm-DC-OSDisk",
+      "vmDataDiskName": "vm-DC-DataDisk",
+      "vmVmSize": "[parameters('vmDCSize')]",
+      "vmNicName": "[concat(variables('generalSettings').vmDCName, '-', 'nic-0')]",
+      "nicPrivateIPAddress": "10.0.1.4",
+      "vmPublicIPName": "vmDC-PublicIP",
+      "vmPublicIPDnsName": "[toLower(replace(concat(resourceGroup().name, '-', variables('generalSettings').vmDCName), '_', '-'))]"
+    },
+    "vmSQL": {
+      "vmImagePublisher": "MicrosoftSQLServer",
+      "vmImageOffer": "SQL2016SP1-WS2016",
+      "vmImageSKU": "Standard",
+      "vmOSDiskName": "vm-SQL-OSDisk",
+      "vmDataDiskName": "vm-SQL-DataDisk",
+      "vmVmSize": "[parameters('vmSQLSize')]",
+      "vmNicName": "[concat(variables('generalSettings').vmSQLName, '-', 'nic-0')]",
+      "vmPublicIPName": "vmSQL-PublicIP",
+      "vmPublicIPDnsName": "[toLower(replace(concat(resourceGroup().name, '-', variables('generalSettings').vmSQLName), '_', '-'))]"
+    },
+    "vmSP": {
+      "vmImagePublisher": "MicrosoftSharePoint",
+      "vmImageOffer": "MicrosoftSharePointServer",
+      "vmImageSKU": "[parameters('sharePointVersion')]",
+      "vmOSDiskName": "vm-SP-OSDisk",
+      "vmDataDiskName": "vm-SP-DataDisk",
+      "vmVmSize": "[parameters('vmSPSize')]",
+      "vmNicName": "[concat(variables('generalSettings').vmSPName, '-', 'nic-0')]",
+      "vmPublicIPName": "vmSP-PublicIP",
+      "vmPublicIPDnsName": "[toLower(replace(concat(resourceGroup().name, '-', variables('generalSettings').vmSPName), '_', '-'))]"
```

Comment:
best not to use the rg name in the dns name - resourceGroup names can contain chars not allowed in a dns name.

Reply:
reply in previous comment